code/game/machinery/recycle_vendor.dm

```diff
@@ -587,13 +633,19 @@
 			highestthreshold = current
 	icon_state = initial(icon_state)+"[highestthreshold]"
 
+/obj/machinery/amesilo/power_change()
+	..()
+	for(var/obj/machinery/amerecycler/toupdate in linked)
+		toupdate.update_icon()
+
 /obj/machinery/amesilo/examine(mob/user, distance, infix, suffix)
 	. = ..()
 	if(get_dist_euclidian(user, src) <= 2)
 		to_chat(user, "[src] is using [copytext_char(icon_state, length(initial(icon_state))+1,0)] out of [SPRITECAPACITY] contained storage modules.")
 
 
 /obj/machinery/amesilo/proc/applybudget()
+	settledebt() // taxes first
```

Comment:
if the associate account is having money transferred to it, taxes are first paid before any money can be transferred to the associate account.